File: old/actions/install_postgresql.py
"""
Install PostgreSQL action
"""
import logging
from cli import Apt
from .base import Action
logger = logging.getLogger(__name__)

class InstallPostgresqlAction(Action):
    """Action to install PostgreSQL package"""
    description = "postgresql installation"

    def execute(self) -> bool:
        """Install PostgreSQL package"""
        if not self.apt_service:
            logger.error("APT service not initialized")
            return False
        # Get version from params
        version = "17"
        if hasattr(self, "_container_cfg") and self._container_cfg:
            params = self._container_cfg.params or {}
            version = str(params.get("version", "17"))
        logger.info("Installing PostgreSQL %s package...", version)
        install_cmd = Apt().install([f"postgresql-{version}", "postgresql-contrib"])
        output = self.apt_service.execute(install_cmd)
        if output is None:
            logger.error("PostgreSQL installation failed")
            return False
        return True

File: src/actions/install_postgresql.go
package actions

import (
	"fmt"
	"enva/libs"
	"enva/services"
)

// InstallPostgresqlAction installs PostgreSQL package
type InstallPostgresqlAction struct {
	*BaseAction
}

func NewInstallPostgresqlAction(sshService *services.SSHService, aptService *services.APTService, pctService *services.PCTService, containerID *string, cfg *libs.LabConfig, containerCfg *libs.ContainerConfig) Action {
	return &InstallPostgresqlAction{
		BaseAction: &BaseAction{
			SSHService:   sshService,
			APTService:   aptService,
			PCTService:   pctService,
			ContainerID: containerID,
			Cfg:          cfg,
			ContainerCfg: containerCfg,
		},
	}
}

func (a *InstallPostgresqlAction) Description() string {
	return "postgresql installation"
}

func (a *InstallPostgresqlAction) Execute() bool {
	if a.APTService == nil {
		libs.GetLogger("install_postgresql").Printf("APT service not initialized")
		return false
	}
	version := "17"
	if a.ContainerCfg != nil && a.ContainerCfg.Params != nil {
		if v, ok := a.ContainerCfg.Params["version"].(string); ok {
			version = v
		}
	}
	libs.GetLogger("install_postgresql").Printf("Installing PostgreSQL %s package...", version)
	output, exitCode := a.APTService.Install([]string{fmt.Sprintf("postgresql-%s", version), "postgresql-contrib"})
	if exitCode == nil || *exitCode != 0 {
		libs.GetLogger("install_postgresql").Printf("PostgreSQL installation failed: %s", output)
		return false
	}
	return true
}

